compiler/semtypes.nim

```diff
@@ -145,7 +145,7 @@ proc semEnum(c: PContext, n: PNode, prev: PType): PType =
 
 proc semSet(c: PContext, n: PNode, prev: PType): PType =
   result = newOrPrevType(tySet, prev, c)
-  if sonsLen(n) == 2:
+  if sonsLen(n) == 2 and n.sons[1].kind != nkEmpty:
```

Comment:
`illFormedAst` is better here since the AST is completely wrong.
Also, missing test case.

Reply:
well the written ast is completely wrong. What the parser generates from it is not that wrong at all. The generated AST just misses a type expression.